Feat: Document Dio for Dart/Flutter (#4857)
getsentry/sentry-docs: src/platforms/dart/common/performance/instrumentation/automatic-instrumentation.mdx

When the HTTP request throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in sentry.io.

For more information see our [SentryHttpClient integration](/platforms/dart/configuration/integrations/http-integration/#performance-monitoring-for-http-requests).

> ### Dio Library Instrumentation

Reviewer: Perhaps easier for those who don't know about Dio, to know this is an HTTP library

```suggestion
### Dio HTTP Library Instrumentation
```

> The Dio instrumentation, once added the Dio integration feature, starts a span out of the active span bound to the scope for each HTTP Request. The SDK sets the span `operation` to `http.client` and `description` to request `$method and $URL`; for example, `GET https://sentry.io`.

Reviewer 2: ```suggestion
The Dio instrumentation, once added the Dio integration feature, starts a span out of the active span bound to the scope for each HTTP Request. The SDK sets the span `operation` to `http.client` and `description` to request `$method $URL`; for example, `GET https://sentry.io`.
```

Reviewer 2: It also provides spans for (de)serialization.

> The Dio instrumentation, once added the Dio integration feature, starts a span out of the active span bound to the scope for each HTTP Request. The SDK sets the span `operation` to `http.client` and `description` to request `$method $URL`; for example, `GET https://sentry.io`.

Reviewer 3: Something is missing in the first part of the first sentence, but I'm not sure what. I've added "to" but that's a guess. I've also rearranged the sentence to improve the flow. 

```suggestion
Once added to the Dio integration feature, the Dio instrumentation starts a span out of the active span bound to the scope for each HTTP request. The SDK sets the span `operation` to `http.client` and the `description` to request `$method $URL`. For example, `GET https://sentry.io`.
```

> The span finishes once the request has been executed. The span `status` depends on either the HTTP Response `code` or `SpanStatus.internalError()` if the `code` does not match any of Sentry's `SpanStatus`.

Reviewer 3: ```suggestion
The span finishes once the request has been executed. The span `status` depends on either the HTTP response `code` or `SpanStatus.internalError()` if the `code` does not match any of Sentry's `SpanStatus` options.
```

> When the HTTP request throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in sentry.io.

Reviewer 3: Should "SentryEvent" be in code format?

> When the HTTP request throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in sentry.io.

Reviewer 3: ```suggestion
When the HTTP request throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in [sentry.io](https://sentry.io).
```

> For more information see our [Dio integration](/platforms/dart/configuration/integrations/dio/#performance-monitoring-for-http-requests).

Reviewer 3: ```suggestion
Learn more in our [Dio integration documentation](/platforms/dart/configuration/integrations/dio/#performance-monitoring-for-http-requests).
```
